pkg/api/v1beta1/clustercurator_types.go

```diff
@@ -79,7 +88,15 @@ type UpgradeHooks struct {
 	// +optional
 	Upstream string `json:"upstream,omitempty"`
 
-	Hooks Hooks `json:"hooks,omitempty"`
+	// Jobs to run before the cluster upgrade
+	Prehook []Hook `json:"prehook,omitempty"`
+
+	// Jobs to run after the cluster upgrade
+	Posthook []Hook `json:"posthook,omitempty"`
+
+	// When provided, this is a Job specification and overrides the default flow
+	// +kubebuilder:pruning:PreserveUnknownFields
+	OverrideJob *runtime.RawExtension `json:"overrideJob,omitempty"`
```

Comment:
this is getting duplicated.. if used Hooks, there is extra layer hooks in resource and towerAuthSecret will be part of Hooks
```
---
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: ClusterCurator
metadata:
  name: leena-ocp
  namespace: leena-ocp
  labels:
    open-cluster-management: curator
spec:
  desiredCuration: upgrade
  upgrade: 
    desiredUpdate: 4.5.13
    hooks:
      towerAuthSecret: "hi"
      posthook:
        - name: Demo Job Template
          extra_vars:
            sn_severity: 1
            sn_priority: 1
```